### PR TITLE
Implementation of platform services for Android

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@chinmaygarde

File: sky/services/platform/src/org/domokit/platform/SystemChromeImpl.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            callback.call(false);
+            return;
+        }
+
+        // Currently the Android implementation only supports masks with zero or one
+        // selected device orientations.
+        int androidOrientation;
+        if (deviceOrientationMask == 0) {
+            androidOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED;
+        } else if (deviceOrientationMask == DeviceOrientation.PORTRAIT_UP) {
```
Comment (chinmaygarde) [inline sky/services/platform/src/org/domokit/platform/SystemChromeImpl.java]: This is a mask. So something like `deviceOrientationMask & DeviceOrientation.PORTRAIT_UP != 0` perhaps

File: sky/services/platform/src/org/domokit/platform/SystemChromeImpl.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        } else if (deviceOrientationMask == DeviceOrientation.PORTRAIT_DOWN) {
+            androidOrientation = ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT;
+        } else if (deviceOrientationMask == DeviceOrientation.LANDSCAPE_RIGHT) {
+            androidOrientation = ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE;
+        } else {
+            callback.call(false);
+            return;
+        }
+
+        activity.setRequestedOrientation(androidOrientation);
```
Comment (chinmaygarde) [inline sky/services/platform/src/org/domokit/platform/SystemChromeImpl.java]: The mojom API specifies the set of valid orientations for the user interface to be in. Not one explicit orientation.

Comment (jason-simmons) [inline sky/services/platform/src/org/domokit/platform/SystemChromeImpl.java]: This is an initial implementation that only supports part of the interface.  Later I can look into allowing transitions among a subset of the orientations.

**Comment by chinmaygarde** [comment]
For the orientation stuff, I suspect you will require a custom orientation event listener sampling at a fairly low rate and set the orientation manually when on update to one that is enabled in the mask.

If you are planning on implementing this in a subsequent patch, lgtm on this. Thanks!

**Comment by chinmaygarde** [comment]
Also, let me know when you land this. I need to patch the iOS platform impl for the mojom changes

**Comment by j9brown** [comment]
NB. I don't recommend writing your own custom orientation listener.  (Full disclosure: I wrote Android's orientation listener.)

**Pull request merged**